lib/spack/spack/database.py

```diff
@@ -604,6 +601,9 @@ def _path(self, spec: "spack.spec.Spec") -> pathlib.Path:
         return self.dir / f"{spec.name}-{spec.dag_hash()}"
 
 
+SelectType = Optional[Callable[[InstallRecord], bool]]
```

Comment:
```suggestion
SelectType = Callable[[InstallRecord], bool]
```

The Optional bit is part of the function signature instead of the type imo.

Reply:
Done in https://github.com/spack/spack/pull/47102/commits/1f8b9967e07189e629636b9998f49b4fc71571b6